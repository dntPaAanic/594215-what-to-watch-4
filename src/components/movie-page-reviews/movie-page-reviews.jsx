import React from 'react';
import PropTypes from 'prop-types';

const MoviePageReviews = (props) => {
  const {film} = props;
  const {comments} = film;
  return (
    <React.Fragment>
      <div className="movie-card__reviews movie-card__row">
        <div className="movie-card__reviews-col">
          {comments && comments.map((comment) => {
            return (
              <div className="review" key={`${comment.author}-${comment.id}`}>
                <blockquote className="review__quote">
                  <p className="review__text">{comment.review}</p>
                  <footer className="review__details">
                    <cite className="review__author">{comment.author}</cite>
                    <time className="review__date" dateTime="2016-12-24">{comment.date}</time>
                  </footer>
                </blockquote>
                <div className="review__rating">{comment.rating}</div>
              </div>
            );
          })}
        </div>
      </div>
    </React.Fragment>
  );
};

MoviePageReviews.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    imagePreview: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseDate: PropTypes.number.isRequired,
    imagePoster: PropTypes.string.isRequired,
    imageBackground: PropTypes.string.isRequired,
    ratingScore: PropTypes.number.isRequired,
    ratingCount: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    previewSrc: PropTypes.string.isRequired,
    runTime: PropTypes.number.isRequired,
    comments: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      review: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired,
      rating: PropTypes.string.isRequired
    })).isRequired,
  }).isRequired
};

export default MoviePageReviews;
