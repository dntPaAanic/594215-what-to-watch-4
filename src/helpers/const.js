const PREVIEW_DELAY = 1000;

const SIMILAR_FILMS_COUNT = 4;

const RatingLevel = {
  BAD: 0,
  NORMAL: 3,
  GOOD: 5,
  VERY_GOOD: 8,
  AWESOME: 10
};

export {PREVIEW_DELAY, SIMILAR_FILMS_COUNT, RatingLevel};
