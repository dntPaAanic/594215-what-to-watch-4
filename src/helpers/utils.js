const PREVIEW_DELAY = 1000;
const RatingLevel = {
  BAD: 0,
  NORMAL: 3,
  GOOD: 5,
  VERY_GOOD: 8,
  AWESOME: 10
};

const getRatingLevel = (ratingScore) => {
  switch (true) {
    case (ratingScore < RatingLevel.NORMAL):
      return `Bad`;
    case (ratingScore < RatingLevel.GOOD):
      return `Normal`;
    case (ratingScore < RatingLevel.VERY_GOOD):
      return `Good`;
    case (ratingScore < RatingLevel.AWESOME):
      return `Very good`;
    case (ratingScore === RatingLevel.AWESOME):
      return `Awesome`;
    default:
      break;
  }
  return null;
};

export {PREVIEW_DELAY, getRatingLevel};
