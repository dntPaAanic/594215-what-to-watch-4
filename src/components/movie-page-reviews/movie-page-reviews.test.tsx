import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {MoviePageReviews} from './movie-page-reviews';
import {film} from '../../helpers/test-data';
import {noop} from '../../helpers/utils';

it(`MoviePageReviews should render correct`, () => {
  const tree = renderer.create(
      <MoviePageReviews
        film={film}
        comments={[]}
        loadComments={noop}
        isCommentsLoaded={true}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
