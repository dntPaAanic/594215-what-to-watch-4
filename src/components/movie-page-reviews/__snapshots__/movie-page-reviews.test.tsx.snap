// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageReviews should render correct 1`] = `
<h2>
  Be first to leave a review!
</h2>
`;
